bloom-housing/bloom
# feat: lottery partners experience

This PR addresses #4047


## Description

Adds most of the partner experience of the lottery, including the following states:

* The card when a lottery has not yet been released to partners
* The card when a lottery has been released but a partner has yet to publish it
* The modal for a partner to publish the lottery
* The card when a lottery has been published

It does not include anything to do with exporting as a partner other than the card content.

## How Can This Be Tested/Reviewed?

Note that the [Figma](https://www.figma.com/design/PAeCVANDmGdTz2Y53ouErM/Lotteries?node-id=3528-20824&t=M2ITZU0BnKLdM5wN-0) is no longer completely accurate. The confirmed copy and design is [here](https://docs.google.com/document/d/130OuxoQnKnukoVr_SPQq57A0JZ4vycdcS9q0EX2Gkso/edit#heading=h.btie0bci2vik) - the copy in text is the confirmed copy, not the copy in the screenshots.

With the toggle on, visit a closed listing with a lottery as a partner, and you should see the [lottery-not-yet-released card](https://docs.google.com/document/d/130OuxoQnKnukoVr_SPQq57A0JZ4vycdcS9q0EX2Gkso/edit#heading=h.57bn0wexwz4x) state.

Visit the listing as an admin and run / release the lottery.

As the partner, you should now see the[ publish-to-public card](https://docs.google.com/document/d/130OuxoQnKnukoVr_SPQq57A0JZ4vycdcS9q0EX2Gkso/edit#heading=h.uguzue28thte), and the [modal](https://docs.google.com/document/d/130OuxoQnKnukoVr_SPQq57A0JZ4vycdcS9q0EX2Gkso/edit#heading=h.ilfx1xnwk3z) when clicking publish.

After clicking publish, you see the [export card](https://docs.google.com/document/d/130OuxoQnKnukoVr_SPQq57A0JZ4vycdcS9q0EX2Gkso/edit#heading=h.qcywz316xi0h) but with no alert.

Some of the dates in the strings will be empty because we don't have history log data yet.

It will now actually change the status from null --> ran --> releasedToParters --> publishedToPublic

## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
